File: frontend/src/lib/api/nomina.ts
import api from './client';

export interface NominaItem {
    fecha: string;
    factura_numero: string;
    item_nombre: string;
    tipo: 'servicio' | 'producto';
    cantidad: number;
    precio_unitario: number;
    subtotal: number;
    comision_porcentaje: number;
    comision_valor: number;
}

export interface NominaResumenEspecialista {
    especialista_id: number;
    nombre: string;
    apellido: string;
    total_servicios: number;
    total_productos: number;
    comision_servicios: number;
    comision_productos: number;
    total_comision: number;
    items?: NominaItem[];
}

export interface NominaResponse {
    fecha_inicio: string;
    fecha_fin: string;
    resumen: NominaResumenEspecialista[];
}

export const nominaService = {
    getResumen: async (fechaInicio: string, fechaFin: string, sedeId?: number) => {
        const params = new URLSearchParams({
            fecha_inicio: fechaInicio,
            fecha_fin: fechaFin,
        });
        if (sedeId) params.append('sede_id', sedeId.toString());

        const response = await api.get<NominaResponse>(`/nomina/resumen?${params.toString()}`);
        return response.data;
    },

    getDetalleEspecialista: async (id: number, fechaInicio: string, fechaFin: string) => {
        const params = new URLSearchParams({
            fecha_inicio: fechaInicio,
            fecha_fin: fechaFin,
        });
        const response = await api.get<NominaResumenEspecialista>(`/nomina/detalle/${id}?${params.toString()}`);
        return response.data;
    }
};

File: backend/app/schemas/nomina.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from decimal import Decimal

class NominaItem(BaseModel):
    fecha: datetime
    factura_numero: str
    item_nombre: str
    tipo: str  # 'servicio' | 'producto'
    cantidad: Decimal
    precio_unitario: Decimal
    subtotal: Decimal
    comision_porcentaje: Decimal
    comision_valor: Decimal
    
    class Config:
        from_attributes = True

class NominaResumenEspecialista(BaseModel):
    especialista_id: int
    nombre: str
    apellido: str
    total_servicios: Decimal  # Monto total vendido en servicios
    total_productos: Decimal  # Monto total vendido en productos
    comision_servicios: Decimal
    comision_productos: Decimal
    total_comision: Decimal
    items: List[NominaItem] = [] # Optional, for detailed view

    class Config:
        from_attributes = True

class NominaResponse(BaseModel):
    fecha_inicio: datetime
    fecha_fin: datetime
    resumen: List[NominaResumenEspecialista]
